src/Resources/contao/modules/ModuleChangePassword.php

```diff
@@ -53,6 +54,12 @@ public function generate()
 			return '';
 		}
 
+		// Require the member to be fully authenticated (see #1273)
+		if (!\System::getContainer()->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY'))
+		{
+			throw new AccessDeniedException('Please authenticate');
```

Comment:
As this message is not shown to the user, I think it should rather be something like `Full authentication is required to change the password`.